06_puppy/alfredxiao/mapstore.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+)
+
+type mapStore struct {
```

Comment:
There's an option here to use promoted fields / embedded types that means you don't need to create constructor functions but there are pros and cons so keep this.

Reply:
surely correct, depending on context